test/exactly-once/exactlyonceck/exactlyonceck

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env bash
+
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Test that a catalog created by materialized v0.1.0 is compatible with the
+# version of materialized under test.
+
+set -euo pipefail
+
+say() {
+    echo "### $* ###"
+}
+
+run_sql() {
+    psql -U materialize -h localhost -p 6875 materialize -c "\pset footer off" -c "$1"
+}
+
+launch_materialized() {
+    name=$1
+    shift
+    "materialized-$name" -w1 "$@" &
+    materialized_pid=$?
+    wait-for-it --timeout=30 -q localhost:6875
+    run_sql "SELECT 1" > /dev/null
+}
+
+kill_materialized() {
+  kill "$materialized_pid"
+  wait 2> /dev/null
+}
+
+wait-for-it --timeout=30 kafka:9092
+wait-for-it --timeout=30 schema-registry:8081
+
+seed=$(od -vAn -N4 -tu4 < /dev/urandom | tr -d '[:space:]')
+
+say "launching materialized-edge"
+launch_materialized edge --log-file=stderr
+
+say "initially populating topics"
+testdrive --kafka-addr kafka:9092 --schema-registry-url=http://schema-registry:8081 --kafka-option=group.id=group1 --seed "$seed" <<'EOF'
+$ set consistency={
+     "name": "materialize.byo.consistency",
+     "type": "record",
+     "fields": [
+         {
+           "name": "source",
+           "type": "string"
+         },
+         {
+           "name": "partition_count",
+           "type": "int"
+         },
+         {
+           "name": "partition_id",
+           "type": ["int","string"]
+         },
+         {
+            "name": "timestamp",
+            "type": "long"
+         },
+         {
+           "name": "offset",
+           "type": "long"
+         }
+     ]
+  }
+
+$ set schema={
+    "type": "record",
+    "name": "envelope",
+    "fields": [
+      {"name": "a", "type": "long"},
+      {"name": "b", "type": "long"}
+    ]
+  }
+
+$ kafka-create-topic topic=input-consistency
+
+$ kafka-ingest format=avro topic=input-consistency schema=${consistency}
+{"source": "testdrive-input-${testdrive.seed}", "partition_count": 1, "partition_id": {"int": 0}, "timestamp": 1, "offset": 4}
```

Comment:
might be good to test with more than 1 input timestamp

Reply:
Will do!